### DISCUSSION TITLE
Support for multi cloud testing and big RPS

### DISCUSSION THREAD

**Discussion by db-dhananjayb** [artilleryio/artillery#2221] [Artillery]
I am impressed by distributed nature of artillery and I want to evaluate this tool with respect to multicloud support. I have some quick questions before getting started:
1. If I correctly understand currently, executing tests on aws is supported. In the vendor support, it also mentions about GCP,(google cload) does it mean one can use artillery inside  our own GCP ?
2. Does it support microsoft Azure? If not Is thee a plan to support microsoft azure with tentative timelines? The documentation mentions something in this regard however I am not sure if Azure support is as extensive as AWS.

I came across similar discussion https://github.com/artilleryio/artillery/discussions/2096 in this context unfortunately it does not mention any timelines.

3. I want to simulate a load of around 10K RPS. Has anyone used it for such a volume? Are there any metrics to refer in this context. ( I came across this blog https://auth0.com/blog/optimizing-cost-and-performance-of-auth0s-private-cloud/ in this context which refers to  6K+ RPS).
Thanks in advance!

**Comment by hassy** [comment]
hi @db-dhananjayb 👋 

We plan to add support for GCP and Azure in the future, but we don't have a timeline yet. We're prioritizing building a first-class experience on AWS first as that's where we see the most interest from our users.

10k RPS is not going to be a problem. This blog post includes a demo of a test running at 100k RPS on AWS Lambda - https://www.artillery.io/blog/this-load-test-cost-us-how-much We've run internal tests that reached 1M RPS. Artillery scales out horizontally so most of the time you just need to increase the number of Lambdas or Fargate containers to increase RPS.
